gui/src/launcher.rs

```diff
@@ -164,6 +225,89 @@ pub enum Message {
 pub enum ViewMessage {
     StartInstall,
     Check(Network),
+    DeleteWallet(DeleteWalletMessage),
+}
+
+#[derive(Debug, Clone)]
+pub enum DeleteWalletMessage {
+    ShowModal(Network),
+    CloseModal,
+    Confirm,
+}
+
+struct DeleteWalletModal {
+    network: Network,
+    wallet_datadir: PathBuf,
+    warning: Option<std::io::Error>,
+    deleted: bool,
+}
+
+impl DeleteWalletModal {
+    fn new(network: Network, wallet_datadir: PathBuf) -> Self {
+        Self {
+            network,
+            wallet_datadir,
+            warning: None,
+            deleted: false,
+        }
+    }
+
+    fn update(&mut self, message: Message) -> Command<Message> {
+        if let Message::View(ViewMessage::DeleteWallet(DeleteWalletMessage::Confirm)) = message {
+            self.warning = None;
+            if let Err(e) = std::fs::remove_dir_all(&self.wallet_datadir) {
```

Comment:
This would remove the bitcoind watchonly wallet file. Is it an issue? Depends. It's certainly not the best to be removing a file managed by a process from under it. On the other hand it's unlikely to make bitcoind crash, and even if it did i don't think it would be worse than any other alternative i can think of for handling this.

Therefore i think it's fine like that, but wanted to share in case other think about something i would have missed.

Reply:
Would it be worth adding a message relating to this in the modal (if they are not using Liana-managed node for that wallet)?

Reply:
I don't think so.